include/ruby/internal/intern/array.h

```diff
@@ -88,6 +88,11 @@ VALUE rb_ary_new(void);
  */
 VALUE rb_ary_new_capa(long capa);
 
+/**
+ * Identical to rb_ary_new_capa(), except it additionally specifies a klass
+ */
+VALUE rb_ary_new_capa_with_klass(VALUE klass, long capa);
```

Comment:
sorry for long absent.
we shouldn't expose new C-API even if Ruby internal requires it.
it should be in internal/array.h.